elasticsearch_dsl/_async/document.py

```diff
@@ -186,14 +229,19 @@ async def mget(
             error_ids = [doc["_id"] for doc in error_docs]
             message = "Required routing not provided for documents %s."
             message %= ", ".join(error_ids)
-            raise RequestError(400, message, error_docs)
+            raise RequestError(400, message, error_docs)  # type: ignore
```

Comment:
@pquentin Forgot to mention this one. I had to type ignore these because the first arg of `RequestError` (which is just an alias for the transport's `APIError`) is typed as a `str`, and I believe not intended to be the status code.

```python
>>> r = RequestError(400, "bad request", {"foo":"bar"})
>>> r.status_code
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/miguelgrinberg/Documents/dev/elasticsearch-dsl-py/venv/lib/python3.12/site-packages/elasticsearch/exceptions.py", line 46, in status_code
    return self.meta.status
           ^^^^^^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'status'
```

Sounds like this is a bug that we should fix at some point.

Reply:
Yes, this is a bug. ApiError is actually quite annoying to raise manually now: https://github.com/elastic/elastic-transport-python/blob/39488817cd5da824101322e40652d17938f0acac/tests/test_exceptions.py#L36-L41